server.js

```diff
@@ -13,16 +14,100 @@ mongoose.Promise = Promise
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
+const listEndpoints = require('express-list-endpoints')
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+//model
+const Thought = mongoose.model("Thought", {
+  message: {
+    type: String,
+    required: [true, "Enter your thought"],
+    minlength: [5, "The text is too short. Must be min 5 characters."],
+    maxlength: [140, "The text is too long. Can be max 140 characters."],
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  // res.send('Hello, welcome to happy thoughts API. Endpoints: /thoughts (methods: GET, POST), /thoughts/:thoughtId/heart (method: POST)')
+  if (res) {
+    res.status(200).send(listEndpoints(app))
+  } else {
+    res.status(404).send("No endpoints found.")
+  }
+
+})
+
+app.get("/thoughts", async (req, res) => {
+
+  try {
+    const thoughts = await Thought.find()
+      .sort({ createdAt: "desc" })
+      .limit(20)
+      .exec()
+    res.status(200).json(thoughts)
+  } catch (err) {
+    res.status(400).json({ message: "Could not get the thoughts", errors: err.errors })
+  }
+})
+
+//endpoint for posting thoughts
+app.post("/thoughts", async (req, res) => {
+
+  try {
+    //retrieve the info sent by the client to our API endpoint
+    const message = req.body.message
+
+    //use mongoose model to create the DB entry
+    const thought = new Thought({ message })
+
+    //save the entry
+    const savedThought = await thought.save()
```

Comment:
Any particular reason why you don't do this in one go? "const thought = new Thought({ message }).save()?

Reply:
Not really. I usually write things in a longer way, since I find it easier to understand, but in this caase I could have written it in asingle line like you pointed out. 👍